segmentation violation after committing deleted files, unable to reproduce so far

**Describe the bug**
Hi, after committing a few deleted files in my repo I got a segmentation violation. I am not quite sure why this happened as I have committed deleted files before and it was working fine.

**To Reproduce**
Unable to reproduce so far.

**Expected behavior**
No segmentation violation

**Screenshots**
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xa95f09]

goroutine 1 [running]:
github.com/jesseduffield/lazygit/pkg/gui/controllers.(*FilesController).canEditFiles.func1(...)
        /home/runner/work/lazygit/lazygit/pkg/gui/controllers/files_controller.go:713
github.com/samber/lo.NoneBy[...](...)
        /home/runner/work/lazygit/lazygit/vendor/github.com/samber/lo/intersect.go:85
github.com/jesseduffield/lazygit/pkg/gui/controllers.(*FilesController).canEditFiles(0x0?, {0xc0009ca068?, 0xaf0745?, 0xc000241940?})
        /home/runner/work/lazygit/lazygit/pkg/gui/controllers/files_controller.go:713 +0x29
github.com/jesseduffield/lazygit/pkg/gui/controllers.(*FilesController).GetKeybindings.(*ListControllerTrait[...]).itemsSelected.func5()
        /home/runner/work/lazygit/lazygit/pkg/gui/controllers/list_controller_trait.go:98 +0xe3
github.com/jesseduffield/lazygit/pkg/gui/controllers.(*FilesController).GetKeybindings.(*ListControllerTrait[...]).require.func6()
        /home/runner/work/lazygit/lazygit/pkg/gui/controllers/list_controller_trait.go:38 +0x47
github.com/jesseduffield/lazygit/pkg/gui/types.(*Binding).IsDisabled(...)
        /home/runner/work/lazygit/lazygit/pkg/gui/types/keybindings.go:47
github.com/jesseduffield/lazygit/pkg/gui.(*OptionsMapMgr).renderContextOptionsMap.func1(0xc0009ca060?, 0x1?)
        /home/runner/work/lazygit/lazygit/pkg/gui/options_map.go:44 +0x27
github.com/samber/lo.Filter[...]({0xc0009d6288, 0x46, 0xc0009ea000}, 0xd78228?)
        /home/runner/work/lazygit/lazygit/vendor/github.com/samber/lo/slice.go:15 +0x9e
github.com/jesseduffield/lazygit/pkg/gui.(*OptionsMapMgr).renderContextOptionsMap(0xc000325390)
        /home/runner/work/lazygit/lazygit/pkg/gui/options_map.go:43 +0x245
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).renderContextOptionsMap(0xc000076858?)
        /home/runner/work/lazygit/lazygit/pkg/gui/options_map.go:27 +0x4e
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).layout(0xc000226608, 0xc00020a8c0)
        /home/runner/work/lazygit/lazygit/pkg/gui/layout.go:168 +0x778
github.com/jesseduffield/gocui.ManagerFunc.Layout(0x0?, 0xc000325518?)
        /home/runner/work/lazygit/lazygit/vendor/github.com/jesseduffield/gocui/gui.go:707 +0x19
github.com/jesseduffield/gocui.(*Gui).flush(0xc00020a8c0)
        /home/runner/work/lazygit/lazygit/vendor/github.com/jesseduffield/gocui/gui.go:1153 +0xab
github.com/jesseduffield/gocui.(*Gui).processEvent(0xc00020a8c0)
        /home/runner/work/lazygit/lazygit/vendor/github.com/jesseduffield/gocui/gui.go:784 +0x2d9
github.com/jesseduffield/gocui.(*Gui).MainLoop(0xc00020a8c0)
        /home/runner/work/lazygit/lazygit/vendor/github.com/jesseduffield/gocui/gui.go:749 +0x9d
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).Run(0xc000226608, {{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}})
        /home/runner/work/lazygit/lazygit/pkg/gui/gui.go:707 +0x6fa
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).RunAndHandleError.func1()
        /home/runner/work/lazygit/lazygit/pkg/gui/gui.go:713 +0x45
github.com/jesseduffield/lazygit/pkg/utils.SafeWithError(0x2a?)
        /home/runner/work/lazygit/lazygit/pkg/utils/utils.go:99 +0x5c
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).RunAndHandleError(0xc000226608, {{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}})
        /home/runner/work/lazygit/lazygit/pkg/gui/gui.go:712 +0xce
github.com/jesseduffield/lazygit/pkg/app.(*App).Run(...)
        /home/runner/work/lazygit/lazygit/pkg/app/app.go:266
github.com/jesseduffield/lazygit/pkg/app.Run({0xe5b5f0?, 0xc0002a28c0?}, 0xc000241180, {{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}})
        /home/runner/work/lazygit/lazygit/pkg/app/app.go:49 +0xd6
github.com/jesseduffield/lazygit/pkg/app.Start(0xc00013df00, {0x0, 0x0})
        /home/runner/work/lazygit/lazygit/pkg/app/entry_point.go:150 +0xd05
main.main()
        /home/runner/work/lazygit/lazygit/main.go:23 +0x98


**Version info:**
lazygit --version
commit=6fcb7eb8bb616c170506312870b3bf15f3dbe37c, build date=2024-05-19T10:15:28Z, build source=binaryRelease, version=0.42.0, os=linux, arch=amd64, git version=2.45.1

I am running wayland, alacritty and tmux if that makes any difference.

Today I faced the same I guess
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xb01d29]

goroutine 1 [running]:
github.com/jesseduffield/lazygit/pkg/gui/controllers.(*FilesController).canEditFiles.func1(...)
        /builddir/build/BUILD/lazygit-0.44.0/pkg/gui/controllers/files_controller.go:724
github.com/samber/lo.NoneBy[...](...)
        /builddir/build/BUILD/lazygit-0.44.0/vendor/github.com/samber/lo/intersect.go:85
github.com/jesseduffield/lazygit/pkg/gui/controllers.(*FilesController).canEditFiles(0x0?, {0xc000128080?, 0xb5e925?, 0xc0001724c0?})
        /builddir/build/BUILD/lazygit-0.44.0/pkg/gui/controllers/files_controller.go:724 +0x29
github.com/jesseduffield/lazygit/pkg/gui/controllers.(*FilesController).GetKeybindings.(*ListControllerTrait[...]).itemsSelected.func5()
        /builddir/build/BUILD/lazygit-0.44.0/pkg/gui/controllers/list_controller_trait.go:98 +0xe3
github.com/jesseduffield/lazygit/pkg/gui/controllers.(*FilesController).GetKeybindings.(*ListControllerTrait[...]).require.func6()
        /builddir/build/BUILD/lazygit-0.44.0/pkg/gui/controllers/list_controller_trait.go:38 +0x47
github.com/jesseduffield/lazygit/pkg/gui/types.(*Binding).IsDisabled(...)
        /builddir/build/BUILD/lazygit-0.44.0/pkg/gui/types/keybindings.go:47
github.com/jesseduffield/lazygit/pkg/gui.(*OptionsMapMgr).renderContextOptionsMap.func1(0xc000128078?, 0x1?)
        /builddir/build/BUILD/lazygit-0.44.0/pkg/gui/options_map.go:44 +0x27
github.com/samber/lo.Filter[...]({0xc00015ca08, 0x48, 0xda949f}, 0xe50d18?)
        /builddir/build/BUILD/lazygit-0.44.0/vendor/github.com/samber/lo/slice.go:15 +0x9e
github.com/jesseduffield/lazygit/pkg/gui.(*OptionsMapMgr).renderContextOptionsMap(0xc0002ed3a0)
        /builddir/build/BUILD/lazygit-0.44.0/pkg/gui/options_map.go:43 +0x248
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).renderContextOptionsMap(0xc0000f04b8?)
        /builddir/build/BUILD/lazygit-0.44.0/pkg/gui/options_map.go:27 +0x4e
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).layout(0xc0001fa008, 0xc0001ba380)
        /builddir/build/BUILD/lazygit-0.44.0/pkg/gui/layout.go:189 +0x765
github.com/jesseduffield/gocui.ManagerFunc.Layout(0x0?, 0xc0003de080?)
        /builddir/build/BUILD/lazygit-0.44.0/vendor/github.com/jesseduffield/gocui/gui.go:726 +0x19
github.com/jesseduffield/gocui.(*Gui).flush(0xc0001ba380)
        /builddir/build/BUILD/lazygit-0.44.0/vendor/github.com/jesseduffield/gocui/gui.go:1179 +0xab
github.com/jesseduffield/gocui.(*Gui).processEvent(0xc0001ba380)
        /builddir/build/BUILD/lazygit-0.44.0/vendor/github.com/jesseduffield/gocui/gui.go:810 +0x2d9
github.com/jesseduffield/gocui.(*Gui).MainLoop(0xc0001ba380)
        /builddir/build/BUILD/lazygit-0.44.0/vendor/github.com/jesseduffield/gocui/gui.go:775 +0xd7
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).Run(0xc0001fa008, {{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}})
        /builddir/build/BUILD/lazygit-0.44.0/pkg/gui/gui.go:846 +0x4fd
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).RunAndHandleError.func1()
        /builddir/build/BUILD/lazygit-0.44.0/pkg/gui/gui.go:852 +0x45
github.com/jesseduffield/lazygit/pkg/utils.SafeWithError(0xc00012d260?)
        /builddir/build/BUILD/lazygit-0.44.0/pkg/utils/utils.go:99 +0x5c
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).RunAndHandleError(0xc0001fa008, {{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}})
        /builddir/build/BUILD/lazygit-0.44.0/pkg/gui/gui.go:851 +0xce
github.com/jesseduffield/lazygit/pkg/app.(*App).Run(...)
        /builddir/build/BUILD/lazygit-0.44.0/pkg/app/app.go:270
github.com/jesseduffield/lazygit/pkg/app.Run({0xf47278?, 0xc0001222c0?}, 0xc00013d640, {{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}})
        /builddir/build/BUILD/lazygit-0.44.0/pkg/app/app.go:48 +0xd6
github.com/jesseduffield/lazygit/pkg/app.Start(0xc000051f00, {0x0, 0x0})
        /builddir/build/BUILD/lazygit-0.44.0/pkg/app/entry_point.go:167 +0xd05
main.main()
        /builddir/build/BUILD/lazygit-0.44.0/main.go:23 +0x98
```

--- a/pkg/gui/controllers/files_controller.go
+++ b/pkg/gui/controllers/files_controller.go
@@ -44,7 +44,7 @@ func (self *FilesController) GetKeybindings(opts types.KeybindingsOpts) []*types
 		{
 			Key:               opts.GetKey(opts.Config.Universal.Select),
 			Handler:           self.withItems(self.press),
-			GetDisabledReason: self.require(self.itemsSelected()),
+			GetDisabledReason: self.require(self.withFileTreeViewModelMutex(self.itemsSelected())),
 			Description:       self.c.Tr.Stage,
 			Tooltip:           self.c.Tr.StageTooltip,
 			DisplayOnScreen:   true,
@@ -91,7 +91,7 @@ func (self *FilesController) GetKeybindings(opts types.KeybindingsOpts) []*types
 		{
 			Key:               opts.GetKey(opts.Config.Universal.Edit),
 			Handler:           self.withItems(self.edit),
-			GetDisabledReason: self.require(self.itemsSelected(self.canEditFiles)),
+			GetDisabledReason: self.require(self.withFileTreeViewModelMutex(self.itemsSelected(self.canEditFiles))),
 			Description:       self.c.Tr.Edit,
 			Tooltip:           self.c.Tr.EditFileTooltip,
 			DisplayOnScreen:   true,
@@ -145,7 +145,7 @@ func (self *FilesController) GetKeybindings(opts types.KeybindingsOpts) []*types
 		{
 			Key:               opts.GetKey(opts.Config.Universal.Remove),
 			Handler:           self.withItems(self.remove),
-			GetDisabledReason: self.require(self.itemsSelected(self.canRemove)),
+			GetDisabledReason: self.withFileTreeViewModelMutex(self.require(self.itemsSelected(self.canRemove))),
 			Description:       self.c.Tr.Discard,
 			Tooltip:           self.c.Tr.DiscardFileChangesTooltip,
 			OpensMenu:         true,
@@ -182,7 +182,7 @@ func (self *FilesController) GetKeybindings(opts types.KeybindingsOpts) []*types
 			Handler:           self.withItems(self.openMergeConflictMenu),
 			Description:       self.c.Tr.ViewMergeConflictOptions,
 			Tooltip:           self.c.Tr.ViewMergeConflictOptionsTooltip,
-			GetDisabledReason: self.require(self.itemsSelected(self.canOpenMergeConflictMenu)),
+			GetDisabledReason: self.require(self.withFileTreeViewModelMutex(self.itemsSelected(self.canOpenMergeConflictMenu))),
 			OpensMenu:         true,
 			DisplayOnScreen:   true,
 		},
@@ -206,6 +206,15 @@ func (self *FilesController) GetKeybindings(opts types.KeybindingsOpts) []*types
 			Tooltip:           self.c.Tr.ExpandAllTooltip,
 			GetDisabledReason: self.require(self.isInTreeMode),
 		},
+	}
+}
+
+func (self *FilesController) withFileTreeViewModelMutex(callback func() *types.DisabledReason) func() *types.DisabledReason {
+	return func() *types.DisabledReason {
+		self.c.Contexts().Files.FileTreeViewModel.RWMutex.RLock()
+		defer self.c.Contexts().Files.FileTreeViewModel.RWMutex.RUnlock()
+
+		return callback()
 	}
 }
 

